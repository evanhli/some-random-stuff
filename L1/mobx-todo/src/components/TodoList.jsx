import React from 'react';
import { uniqueId } from 'lodash';

const TodoList = ({ todos, status }) => {
  return (
    <ul aria-label='todo-list'>
      { status === 'error' ? <div aria-label='error-add'>ERROR</div> : null}
      { status === 'pending' ? <div aria-label='pending-add'>LOADING</div> : null}
      {
        Array.isArray(todos) ? todos.map((todo) => {
          return <li aria-label='todo-item' key={uniqueId()}>{todo}</li>;
        }) : null
      }
    </ul>
  );
};

export default TodoList;
