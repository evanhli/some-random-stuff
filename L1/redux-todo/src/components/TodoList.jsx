import React from 'react';
import { connect } from 'react-redux';


function mapStateToProps(state) {
  return { todos: state.todos.list, status: state.todos.status };
}

const TodoList = ({ todos, status }) => {
  return (
    <ul aria-label='todo-list'>
      { status === 'error' ? <div>ERROR</div> : null}
      { status === 'pending' ? <div>LOADING</div> : null}
      {
        Array.isArray(todos) ? todos.map((todo) => {
          return <li aria-label='todo-item' key={todo.id}>{todo.text}</li>;
        }) : null
      }
    </ul>
  );
};

export default connect(mapStateToProps)(TodoList);
