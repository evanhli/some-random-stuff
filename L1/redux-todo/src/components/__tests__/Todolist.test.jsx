import React from 'react';
import renderer from 'react-test-renderer';

import TodoList from '../TodoList';

test('renders an item', () => {
  const todos = ['abc', 'def', 'foo', 'bar'];
  const list = renderer.create(<TodoList todos={todos} status='added' />);
  expect(list).toMatchSnapshot();
});
