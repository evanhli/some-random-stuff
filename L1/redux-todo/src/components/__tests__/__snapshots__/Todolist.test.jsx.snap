// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an item 1`] = `
<ul
  aria-label="todo-list"
>
  <li
    aria-label="todo-item"
  >
    abc
  </li>
  <li
    aria-label="todo-item"
  >
    def
  </li>
  <li
    aria-label="todo-item"
  >
    foo
  </li>
  <li
    aria-label="todo-item"
  >
    bar
  </li>
</ul>
`;
